dali/test/python/test_operator_readers_webdataset_corner.py

```diff
@@ -0,0 +1,378 @@
+            batch_size=test_batch_size,
+            device_id=0,
+            num_threads=1,
+        ),
+        test_batch_size,
+        math.ceil(num_samples / test_batch_size) * 10,
+    )
+    wds_pipeline = webdataset_raw_pipeline(
+        tar_file_path,
+        index_file.name,
+        ["txt"],
+        batch_size=test_batch_size,
+        device_id=0,
+        num_threads=1,
+    )
+    wds_pipeline.build()
+    assert_equal(list(wds_pipeline.epoch_size().values())[0], num_samples)
+
+
+def test_mmap_dtype_incompatibility():
+    assert_raises(RuntimeError, corner_case, dtypes=[dali.types.INT8, dali.types.FLOAT64])
+
+
+def test_dont_use_mmap():
+    corner_case(dont_use_mmap=True)
+
+
+def test_skip_cached_images():
+    corner_case(skip_cached_images=True)
+
+
+def test_pad_last_batch():
+    corner_case(pad_last_batch=True)
+
+
+def test_lazy_init():
+    corner_case(lazy_init=True)
+
+
+def test_read_ahead():
+    corner_case(read_ahead=True)
+
+
+def test_argument_errors():
+    def uris_configs_error():
+        webdataset_pipeline = webdataset_raw_pipeline(
+            [
+                get_dali_extra_path() + "/db/webdataset/MNIST/devel-0.tar",
+                get_dali_extra_path() + "/db/webdataset/MNIST/devel-1.tar",
+                get_dali_extra_path() + "/db/webdataset/MNIST/devel-2.tar",
+            ],
+            [],
+            ["jpg", "cls"],
+            batch_size=1,
+            device_id=0,
+            num_threads=1,
+        )
+        webdataset_pipeline.build()
+
+    assert_raises(RuntimeError, uris_configs_error)
+
+    assert_raises(RuntimeError, corner_case, missing_component_behavior="SomethingInvalid")
+    corner_case(missing_component_behavior="Skip")
+
+    assert_raises(RuntimeError, corner_case, dtypes=[dali.types.STRING, dali.types.STRING])
+    assert_raises(RuntimeError, corner_case, dtypes=dali.types.INT8)
+
+
+def test_index_errors():
+    def less_components_error():
+        index_file = tempfile.NamedTemporaryFile()
+        index_file.write(b"512 100\n")
+        index_file.flush()
+        webdataset_pipeline = webdataset_raw_pipeline(
+            get_dali_extra_path() + "/db/webdataset/MNIST/devel-0.tar",
+            index_file.name,
+            ["jpg", "cls"],
+            batch_size=1,
+            device_id=0,
+            num_threads=1,
+        )
+        webdataset_pipeline.build()
+
+    assert_raises(RuntimeError, less_components_error)
+
+    def no_extensions_error():
+        index_file = tempfile.NamedTemporaryFile()
+        index_file.write(b"512 1\n0\n")
+        index_file.flush()
+        webdataset_pipeline = webdataset_raw_pipeline(
+            get_dali_extra_path() + "/db/webdataset/MNIST/devel-0.tar",
+            index_file.name,
+            ["jpg", "cls"],
+            batch_size=1,
+            device_id=0,
+            num_threads=1,
+        )
+        webdataset_pipeline.build()
+
+    assert_raises(RuntimeError, no_extensions_error)
```

Comment:
Again, please check error message using nose_utils.

Reply:
done, will be pushed soon